google-cloud-firestore/src/main/java/com/google/cloud/firestore/Firestore.java

```diff
@@ -85,6 +85,31 @@
   @Nonnull
   <T> ApiFuture<T> runTransaction(@Nonnull final Transaction.Function<T> updateFunction);
 
+  /**
+   * Executes the given updateFunction and then attempts to commit the changes applied within the
+   * transaction. If any document read within the transaction has changed, the updateFunction will
+   * be retried. If it fails to commit after 5 attempts, the transaction will fail.
+   *
+   * @param updateFunction The function to execute within the transaction context.
+   * @return An ApiFuture that will be resolved with the result from updateFunction.
+   */
+  @Nonnull
+  <T> ApiFuture<T> runTransactionAsync(@Nonnull final Transaction.AsyncFunction<T> updateFunction);
```

Comment:
We use the "async" suffix in other parts of the API to indicate that the function returns an ApiFuture. "Async" is usually coupled with a blocking API that lacks the suffix. I think what you are doing here is slightly different - `runTransaction()` is also async, but in a slightly different way.

What do you think of renaming these new methods to `runAsyncTransaction`?

Reply:
method renamed